pkg/cmd/admin/migrate/authorization/authorization.go

```diff
@@ -0,0 +1,262 @@
+package authorization
+
+import (
+	"errors"
+	"fmt"
+	"io"
+
+	apiequality "k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/kubernetes/pkg/apis/rbac"
+	rbacinternalversion "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset/typed/rbac/internalversion"
+	kcmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/kubectl/resource"
+
+	authorizationapi "github.com/openshift/origin/pkg/authorization/api"
+	"github.com/openshift/origin/pkg/authorization/controller/authorizationsync"
+	"github.com/openshift/origin/pkg/cmd/admin/migrate"
+	"github.com/openshift/origin/pkg/cmd/templates"
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+
+	"github.com/spf13/cobra"
+)
+
+var (
+	internalMigrateAuthorizationLong = templates.LongDesc(`
+		Check for parity between Openshift authorization objects and Kubernetes RBAC
+
+		A controller is used to keep Openshift authorization objects and Kubernetes RBAC in sync.
+		This command checks for parity between those objects across all namespaces and reports
+		all objects that are out of sync.  These objects require manual intervention to sync
+		as the controller handles all cases where automatic sync is possible.
+
+		The following resource types are checked by this command:
+
+		* clusterrole
+		* role
+		* clusterrolebinding
+		* rolebinding
+
+		No resources are mutated.`)
+
+	notChanged = migrate.ReporterBool(false)
+
+	errOutOfSync = errors.New("is not in sync with RBAC")
+)
+
+type MigrateAuthorizationOptions struct {
+	migrate.ResourceOptions
+	rbac rbacinternalversion.RbacInterface
+}
+
+func NewCmdMigrateAuthorization(name, fullName string, f *clientcmd.Factory, in io.Reader, out, errout io.Writer) *cobra.Command {
+	options := &MigrateAuthorizationOptions{
+		ResourceOptions: migrate.ResourceOptions{
+			In:            in,
+			Out:           out,
+			ErrOut:        errout,
+			AllNamespaces: true,
+			Include:       []string{"clusterrole", "role", "clusterrolebinding", "rolebinding"},
```

Comment:
`clusterrole.authorization.openshift.io`, right?  This could never successfully be run on a prior version since the RBAC side would be missing.